BinaryObjectScanner.Protection/Macrovision.cs

```diff
@@ -228,13 +277,162 @@ internal static string Get00000001TMPVersion(string firstMatchedString, IEnumera
+                // This file is not currently known to be used in versions past 4.70.000.
+                default:
+                    return "/ Product Unknown (Report this to us on GitHub)";
+            }
+        }
+
+        // TODO: Combine with filesize version checks if possible.
+        private string GetSecDrvExecutableVersion(PortableExecutable pex)
+        {
+            // Different versions of this driver correspond to different SafeDisc versions.
+            // TODO: Check if earlier versions of this driver contain the version string in a less obvious place. 
+            string version = pex.FileVersion;
+            if (!string.IsNullOrEmpty(version))
+            {
+                switch (version)
+                {
+                    // Found to be in Redump entry 32783.
+                    // The product version is "3.17.000 Windows NT 2002/07/01".
+                    case "3.17.000":
+                        return "3.17.000 / SafeDisc 2.80.010-2.80.011";
+                    // Found to be in Redump entry 52606.
+                    // The product version is "3.18.000 Windows NT 2002/11/14".
+                    case "3.18.000":
+                        return "3.18.000 / SafeDisc 2.90.010-2.90.040";
+                    // Found to be in Redump entry 13230.
+                    // The product version is "3.19.000 Windows NT/2K/XP 2003/03/19".
+                    case "3.19.000":
+                        return "3.19.000 / SafeDisc 3.10.020-3.15.011";
+                    // Found to be in Redump entry 58625.
+                    // The product version is "SECURITY Driver 3.22.000 2004/01/16".
+                    case "3.22.000":
+                        return "3.22.000 / SafeDisc 3.20.020-3.20.022";
+                    // Found to be in Redump entry 15614.
+                    // The product version is "SECURITY Driver 4.00.060 2004/08/31".
+                    case "4.00.060":
+                        return "4.00.060 / SafeDisc 4.00.000-4.70.000";
+                    // Found distributed online, but so far not in a game release. TODO: Discover original source.
+                    // Can be found at https://github.com/ericwj/PsSecDrv/blob/master/tools/SECDRV/SECDRV.sys, and the file is confirmed to be distributed officialy by Microsoft: https://www.virustotal.com/gui/file/34bbb0459c96b3de94ccb0d73461562935c583d7bf93828da4e20a6bc9b7301d/.
+                    // The product version is "SECURITY Driver 4.03.086 2006/09/13".
+                    case "4.03.086":
+                        return "4.03.086 / Unknown SafeDisc version";
+                    default:
+                        return $"Unknown Version {version} (Report this to us on GitHub)";
+                }
+            }
+
+            return "Unknown Version (Report this to us on GitHub)";
+        }
+
+        private string CheckNewSectionForProtection(string file, bool includeDebug, List<string> sectionStrings, byte[] sectionRaw)
```

Comment:
What was the reasoning behind splitting this near-identical code out? Are there cases that both the padding and the section contain `BoG` but only one has a version?

Reply:
As per a discussion in Discord, make a new parameter on the original method and gate the functionality behind that instead.